drivers/serial/uart_ns16550.c

```diff
@@ -1248,6 +1248,15 @@ static const struct uart_driver_api uart_ns16550_driver_api = {
 #define DEV_DATA_DLF_INIT(n) \
 	_CONCAT(DEV_DATA_DLF, DT_INST_NODE_HAS_PROP(n, dlf))(n)
 
+#ifdef CONFIG_UART_NS16550_AUTO_INIT_LEVEL
+#define NS16550_BOOT_LEVEL0 PRE_KERNEL_1
+#define NS16550_BOOT_LEVEL1 POST_KERNEL
```

Comment:
Why `POST_KERNEL`? As long as it is initialized after PCIe during `PRE_KERNEL_*`, it will still works.

Reply:
TBH, I don't like this all-or-none approach. Could you a per device tree node approach, where each node would delay initialization based on some device tree property? (for example, add a `pcie-acpi` property to such relation.)

Reply:
> TBH, I don't like this all-or-none approach. Could you a per device tree node approach, where each node would delay initialization based on some device tree property? (for example, add a `pcie-acpi` property to such relation.)

Current changes is per instance (node) and will check device node is on a pcie parent node or not.